igvm/hypervisor.py

```diff
@@ -796,7 +828,7 @@ def redefine_vm(self, vm, new_fqdn=None):
         # XXX: undefine_vm depends on vm.fqdn beeing the old name for finding
         # legacy domains w/o an uid_name.  The order is therefore important.
         vm.fqdn = new_fqdn or vm.fqdn
-        self.define_vm(vm)
+        self.define_vm(vm=vm, vg_name=vm.vg_name)
```

Comment:
```suggestion
        self.define_vm(vm)
```

`define_vm()` has no vg_name argument.
You can also fetch the value inside the function directly with `vm.vg_name`.